mautic/mautic
# Added timezone and preferred_locale to JS tracking event

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR added option to set timezone and preferred_locale from JS tracking code.

Timezone detection taken from https://github.com/Valve/fingerprintjs2/blob/master/fingerprint2.js#L443


#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Set timezone and preferred_locale  custom fields as Publicly updatable
![image](https://user-images.githubusercontent.com/462477/70453620-5ec53880-1aa9-11ea-851b-c2eaf5f869c2.png)
3. Then add tracking code to your page
4. Check If these fields were update based on your browser settings

![image](https://user-images.githubusercontent.com/462477/70453774-9d5af300-1aa9-11ea-8099-41f96d9961c6.png)
